### ISSUE TITLE
Add Linear Regression

### ISSUE THREAD

**Issue by param087** [param087/swiftML#4]
### **Linear Regression:** 
***
#### `SingularValueDecompositionLinearRegression(fitIntercept: true)`
  - **fitIntercept**: Whether to calculate the intercept for this model. If set to `false`, no intercept will be used in calculations.

#### API Methods

  **`fit(data: Tensor, labels: Tensor)`**:  Fit a linear model.
  **`prediction(for: Tensor)`**: Prediction using a linear model.
  **`score(data: Tensor, labels: Tensor)`**: Returns the coefficient of determination `R^2` for the prediction.

***

#### `GradientDescentLinearRegression(iterationCount: 1000, learningRate: 0.001, fitIntercept: true)`
  - **iterationCount**: The maximum number of passes over the training data.
  - **learningRate**: The amount that the weights are updated during training.
  - **fitIntercept**: Whether to calculate the intercept for this model. If set to `false`, no intercept will be used in calculations.

#### API Methods

 **`fit(data: Tensor, labels: Tensor)`**:  Fit a linear model.
  **`prediction(for: Tensor)`**: Prediction using a linear model.
  **`score(data: Tensor, labels: Tensor)`**: Returns the coefficient of determination `R^2` for the prediction.
  
###  References 
 - https://en.wikipedia.org/wiki/Linear_regression
 - https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.LinearRegression.html
 - https://en.wikipedia.org/wiki/Moore%E2%80%93Penrose_inverse
 - https://github.com/eriklindernoren/ML-From-Scratch/blob/master/mlfromscratch/supervised_learning/regression.py
